server/raft.go

```diff
@@ -387,7 +387,7 @@ func (s *Server) startRaftNode(accName string, cfg *RaftConfig, labels pprofLabe
 		apply:    newIPQueue[*CommittedEntry](s, qpfx+"committedEntry"),
 		stepdown: newIPQueue[string](s, qpfx+"stepdown"),
 		accName:  accName,
-		leadc:    make(chan bool, 1),
+		leadc:    make(chan bool, 32),
```

Comment:
Wonder if it is better for this to be an `*ipQueue[bool]` instead? I'd like to hope that we don't switch ever leadership state that quickly 32 times, but if it's important that we never lose that transition, IP queues don't have backpressure like plain channels do.

Reply:
In my experimentation with the `TestStreamSourcingScalingSourcingManyBenchmark` test it only flipped quickly once in a row. But it could certainly be possible that in other scenarios it could flip quickly more than once, but 32 is a pretty high number so (at least in my testing) it doesn't ever get that high and the limit (where signals start getting dropped) is never reached.

Note that in the previous version (#5496) I had not touched the size of the channel and instead made `setLeader()` remember what state it is in and force a cleanup and re-setup in case it was already the leader and it gets a setLeader(true) (which would happen when a signal of false was followed a fraction of a second later with a signal of true - and therefore the signal of false was dropped by the channel because of capacity of 1). But this is not there anymore in this version of the PR.